eclipse-scout-core/src/session/Session.ts

```diff
@@ -1308,68 +1304,21 @@ export class Session extends EventEmitter implements SessionModel, ModelAdapterL
     }
   }
 
-  setBusy(busy: boolean) {
-    if (busy) {
-      if (!this._busy) {
-        this._renderBusy();
-      }
-      this._busy = true;
-    } else {
-      if (this._busy) {
-        this._removeBusy();
-      }
-      this._busy = false;
-    }
-  }
-
-  protected _renderBusy() {
-    if (this._busyIndicatorTimeoutId !== null && this._busyIndicatorTimeoutId !== undefined) {
-      // Do not schedule it twice
-      return;
-    }
-    // Don't show the busy indicator immediately. Set a short timer instead (which may be
-    // cancelled again if the busy state returns to false in the meantime).
-    this._busyIndicatorTimeoutId = setTimeout(() => {
-      if (this._busyIndicator) {
-        // busy indicator is already showing
-        return;
-      }
-      if (!this.desktop || !this.desktop.rendered) {
-        return; // No busy indicator without desktop (e.g. during shutdown)
-      }
-      this._busyIndicator = scout.create(BusyIndicator, {
-        parent: this.desktop
-      });
-      this._busyIndicator.on('cancel', this._onCancelProcessing.bind(this));
-      this._busyIndicator.render(this.$entryPoint);
-    }, 500);
+  /** @see BusySupport.setBusy */
+  setBusy(busy: boolean | BusyIndicatorOptions) {
+    this.desktop?.setBusy(busy);
   }
 
-  protected _removeBusy() {
-    // Clear pending timer
-    clearTimeout(this._busyIndicatorTimeoutId);
-    this._busyIndicatorTimeoutId = null;
-
-    // Remove busy indicator (if it was already created)
-    if (this._busyIndicator) {
-      this._busyIndicator.destroy();
-      this._busyIndicator = null;
-    }
-  }
-
-  protected _onCancelProcessing(event: Event) {
-    let busyIndicator = this._busyIndicator;
-    if (!busyIndicator) {
-      return; // removed in the meantime
-    }
-    busyIndicator.off('cancel');
-
-    // Set "canceling" state in busy indicator (after 100ms, would not look good otherwise)
-    setTimeout(() => {
-      busyIndicator.cancelled();
-    }, 100);
-
-    this._sendCancelRequest();
+  protected _setBusy(busy: boolean) {
```

Comment:
I wanted to keep the Session.setBusy method because it already existed and is still used in various places. So one migration task can be omitted. Furthermore in various places writing this.session.setBusy seems to be better than this.session.desktop?.setBusy which includes the risk of a NPE.
Actually the busy methods on the Session have nothing to do with properties. Therefore I don't see any confusion here. To make that clear I renamed the protected method.